desktop/core/src/desktop/js/apps/storageBrowser/StorageFilePage/StorageFilePage.tsx

```diff
@@ -20,98 +20,175 @@ import './StorageFilePage.scss';
 import { i18nReact } from '../../../utils/i18nReact';
 import Button, { PrimaryButton } from 'cuix/dist/components/Button';
 import { getFileMetaData } from './StorageFilePage.util';
-import { DOWNLOAD_API_URL } from '../../../reactComponents/FileChooser/api';
+import { DOWNLOAD_API_URL, SAVE_FILE_API_URL } from '../../../reactComponents/FileChooser/api';
 import huePubSub from '../../../utils/huePubSub';
+import useSaveData from '../../../utils/hooks/useSaveData';
+import {
+  EDITABLE_FILE_FORMATS,
+  SUPPORTED_FILE_EXTENSIONS,
+  SupportedFileTypes
+} from '../../../utils/constants/storageBrowser';
+import { Spin } from 'antd';
 
 const StorageFilePage = ({ fileData }: { fileData: PathAndFileData }): JSX.Element => {
   const { t } = i18nReact.useTranslation();
   const [isEditing, setIsEditing] = React.useState(false);
   const [fileContent, setFileContent] = React.useState(fileData.view?.contents);
   const fileMetaData = useMemo(() => getFileMetaData(t, fileData), [t, fileData]);
 
+  const { loading: isSaving, save } = useSaveData(SAVE_FILE_API_URL, {
+    onSuccess: () => {
+      setIsEditing(false);
```

Comment:
I would disable editing as soon as the save button was clicked, not when successful. If save failed I would enable the editing again.

Reply:
Make sense. Noted and fixed.